### DISCUSSION TITLE
Trying to connect Zigbee Bridge Pro with Tasmota to Home Assistant through MQTT.

### DISCUSSION THREAD

**Discussion by rsilves** [arendst/Tasmota#18916] [Support request]
Hi everyone. I have a Home Assistant Yellow and most of my Zigbee devices are connected directly to its internal antenna, but I have three devices that won't connect or that don't play nice with that radio, so I am using a spare ZBBridgePro to use these devices (FYI these are a Philips Filament bulb, and two Tuya remotes).
I have been using these devices connected to the Zigbee Bridge Pro radio, and the ZBBridge connected to Home Assistant through ZHA. This worked fine and I've been using it for months. But as HA doesn't let you use two ZHA instances, I was using my internal Yellow antenna with ZigbeeqMQTT and things were sometimes working weird there. So, as I prefer to have my most important devices connected in the most "standard" way (ZHA), I switched things: I am now using the Yellow radio with ZHA, and I'm having trouble to connect the ZBBridgePro in an alternative way.
I read that some people made the ZBBridgrPro work with Zigbee2MQTT using the TCP mode (the same mode I used when it was connected to ZHA).  But I tried different setups and cannot make Zigbee2MQTT start when connected to the ZBBridgrPro.

I also tried setting the ZBBridgePro in the not-TCP mode, the one that shows the "Zigbee Permit Join" button. I paired my devices to the ZBBridgePro using this button and set the MQTT IP/port/username/pass but I don't see the devices in the MQTT explorer. 

![image](https://github.com/arendst/Tasmota/assets/40880011/d6ca6c6d-c92b-44e5-b4c2-0618b4d4f54f)

Do I need to do anything extra in the web interface to make these devices "published" to MQTT?

Does anyone recommend another configuration for the thing I'm trying to do? I just want an easy way to connect these three devices to HA :)

Thanks!
Rodrigo

**Comment by s-hadinger** [comment]
If you are using HA, the obvious choice is to use ZHA.

Zigbee is automatically published to MQTT, as long as you have an MQTT connection up.

**Reply by rsilves** [reply]
Thanks @s-hadinger for your reply, but unfortunately your solution conflicts with what I explained in the post: Home Assistant doesn't admit two instances of ZHA running at the same time. In my scenario, I have ZHA configured with the Home Assistant Yellow internal Zigbee radio, and I have to set up the ZBBridge separately, either connecting it directly to MQTT or by using Zigbee2MQTT (I'm open to more ideas though, I have not had success with any of these options).

When set the MQTT address in the ZBBridge the same way as I have set up my other Tasmota-flashed devices (Sonoff switches), I don't see the devices paired to the ZBBridge in my MQTT explorer.

Could anyone propose a workaround or any tests that could help me find the issue here? Either with the Zigbee2MQTT add-on or with the devices not being published to MQTT.

Thanks!
Rodrigo.

**Reply by s-hadinger** [reply]
I confirm that Zigbee message are automatically published to MQTT. The only reason I see for not receiving them is that you didn't subscribed to them. Could you try subscribing to all messages with `#`?

**Reply by sfromis** [reply]
Specifically, when you in the console get messages tagged with `MQT:`, this tells what gets published via MQTT, like:
`23:43:53.802 MQT: xxxxxxxx/tele/SENSOR = {"Aqara-Cube":{"Device":......`
This shows both the topic and the payload.

**Reply by rsilves** [reply]
Great! I used # to listen to all topics and found that the MQTT server in HA is receiving the messages from the ZBBridge. I guess that, to be able to see and control this device from HA, I would need to add it to the mqtt.yaml file, is that correct?
I have some switches configured over MQTT but they are way simpler, so I need some help here.
Could anyone point me to a configuration example so I can try that? The device I'm trying to configure is Hue Filament Bulb which is dimmable but has no RGB controls.

Thanks,
Rodrigo
